packages/backend/src/managers/playground/McpServerManager.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('McpServerManager', () => {
+  let appUserDirectory: string;
+  let consoleErrors: string;
+  let consoleWarnings: string;
+  beforeEach(async () => {
+    appUserDirectory = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'mcp-server-manager-test-'));
+    consoleErrors = '';
+    vi.spyOn(console, 'error').mockImplementation((...args) => {
+      consoleErrors += args.join(' ') + '\n';
+    });
+    consoleWarnings = '';
+    vi.spyOn(console, 'warn').mockImplementation((...args) => {
+      consoleWarnings += args.join(' ') + '\n';
+    });
+  });
+  afterEach(async () => {
+    await fs.promises.rm(appUserDirectory, { recursive: true });
+  });
+  describe('load', () => {
+    test('with no file, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toContain('McpServerManager: Failed to load MCP settings');
+      expect(consoleWarnings).toBe('');
+    });
+    test('with empty file, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      const mcpSettingsFile = path.join(appUserDirectory, 'mcp-settings.json');
+      await fs.promises.writeFile(mcpSettingsFile, '{}');
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toBe('');
+      expect(consoleWarnings).toBe('');
+    });
+    test('with invalid JSON, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      const mcpSettingsFile = path.join(appUserDirectory, 'mcp-settings.json');
+      await fs.promises.writeFile(mcpSettingsFile, '{invalid json}');
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toContain('McpServerManager: Failed to load MCP settings');
+      expect(consoleWarnings).toBe('');
+    });
+    describe('with valid JSON', () => {
+      let mcpServers: McpServer[];
+      beforeEach(async () => {
+        const mcpSettingsFile = path.join(appUserDirectory, 'mcp-settings.json');
+        const mcpSettings = {
+          servers: {
+            'stdio-ok': {
+              enabled: true,
+              type: 'stdio',
+              command: 'npx',
+              args: ['-y', 'kubernetes-mcp-server'],
+            },
+            'sse-ok': {
+              enabled: true,
+              type: 'sse',
+              url: 'https://echo.example.com/sse',
+              headers: {
+                foo: 'bar',
+              },
+            },
+            'invalid-type': {
+              enabled: true,
+              type: 'invalid',
+              url: 'https://echo.example.com/sse',
+            },
+          },
+        };
+        fs.writeFileSync(mcpSettingsFile, JSON.stringify(mcpSettings));
```

Comment:
Instead of writing the file, we should mock the `node:fs/promise` package, and mock the readFile method used by the MCP server manager